cnm/ipam/ipam.go

```diff
@@ -267,3 +268,35 @@ func (plugin *ipamPlugin) releaseAddress(w http.ResponseWriter, r *http.Request)
 
 	log.Response(plugin.Name, &resp, err)
 }
+
+// Handles retrieving info about ipaddress statistics
+func (plugin *ipamPlugin) getPoolInfo(w http.ResponseWriter, r *http.Request) {
+	var req getPoolInfoRequest
+
+	// Decode request.
+	err := plugin.Listener.Decode(w, r, &req)
+	log.Request(plugin.Name, &req, err)
+	if err != nil {
+		return
+	}
+
+	// Process request.
+	poolId, err := ipam.NewAddressPoolIdFromString(req.PoolID)
+	if err != nil {
+		plugin.SendErrorResponse(w, err)
+		return
+	}
+
+	addrPoolInfo, err := plugin.am.GetPoolInfo(poolId.AsId, poolId.Subnet)
+	if err != nil {
+		plugin.SendErrorResponse(w, err)
+		return
+	}
+
+	// Encode response.
+	resp := getPoolInfoResponse{Capacity: addrPoolInfo.Capacity, Available: addrPoolInfo.Available, UnhealthyAddresses: addrPoolInfo.UnhealthyAddr}
```

Comment:
even requestPoolResponse, requestAddressResponse is also like that. do you want me to change that also or this format is only for get APIs?